src/lib.rs

```diff
@@ -247,7 +287,9 @@ impl Type {
         match (self, other) {
             (_, Type::Unknown) => true,
             (Type::List(a), Type::List(b)) => a.fits(b),
-            (Type::Union(a), Type::Union(b)) => a.iter().all(|x| b.contains(x)),
```

Comment:
I changed this to be stricter. Since all possible values we can get should be valid according to the type.